### PR TITLE
migration: remove claims pallet

### PR DESCRIPTION
**Pull request by wischli**
# Description

On Monday, 3:01pm CET, the claiming of Tinlake rewards was disabled [as a result of CP81](https://gov.centrifuge.io/t/cp81-unclaimed-tinlake-rewards/5885) by uploading an unclaimable claim Merkle Tree root hash. This just serves as a temporary fix such that we need to remove the claims pallet which is done in this PR.

* Removes `pallet-claims` used for claiming tinlake rewards from repository and runtimes
* Adds migration to `centrifuge-runtime` which removes all storage entries associated with `Claims` module

## try-runtime execution

```
[2024-01-31T13:03:08Z INFO  runtime_common::migrations::nuke] Clear-Claims: nuking pallet prefix...
[2024-01-31T13:03:08Z INFO  runtime_common::migrations::nuke] Clear-Claims: storage cleared successful
[2024-01-31T13:03:08Z INFO  runtime_common::migrations::nuke] Clear-Claims: iteration result. backend: 402 unique: 402 loops: 402
```

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: runtime/common/src/migrations/nuke.rs [hunk 1]
```diff
@@ -24,19 +24,93 @@ use sp_runtime::DispatchError;
 #[cfg(feature = "try-runtime")]
 use sp_std::vec::Vec;
 
+/// This upgrade nukes all storage from the pallet individually.
+///
+/// If the pallet shall be kept, please use [ResetPallet] instead because
+/// here we neither check nor reset the corresponding storage version.
+pub struct KillPrefix<PalletName, DbWeight>(sp_std::marker::PhantomData<(PalletName, DbWeight)>);
```
Comment (wischli) [inline runtime/common/src/migrations/nuke.rs:31]: NOTE: I tried reusing the existing `ResetPallet` migration. Unfortunately this requires the pallet to still be defined for the runtime which we don't want. The solution would have been super hacky, so I decided to create a separate migration which nukes a pallet based on the pallet name.

File: runtime/development/src/lib.rs [hunk 2]
```diff
@@ -1900,7 +1883,7 @@ construct_runtime!(
 		// our pallets part 1
 		Fees: pallet_fees::{Pallet, Call, Storage, Config<T>, Event<T>} = 90,
 		Anchor: pallet_anchors::{Pallet, Call, Storage} = 91,
-		Claims: pallet_claims::{Pallet, Call, Storage, Event<T>} = 92,
+		// Removed: Claims = 02
```
Comment (lemunozm) [inline runtime/development/src/lib.rs:1886]: Good gap reminder

File: runtime/common/src/migrations/nuke.rs (hunk 1 above)
Comment (lemunozm) [inline runtime/common/src/migrations/nuke.rs:31]: NIT. Maybe `KillPallet` instead, to be according to the `ResetPallet` name?

File: runtime/centrifuge/src/lib.rs [hunk 3]
```diff
@@ -1139,25 +1139,6 @@ impl chainbridge::Config for Runtime {
 	type WeightInfo = ();
 }
 
-// Parameterize claims pallet
-parameter_types! {
-	pub const ClaimsPalletId: PalletId = cfg_types::ids::CLAIMS_PALLET_ID;
```
Comment (lemunozm) [inline runtime/centrifuge/src/lib.rs:1144]: Could `CLAIMS_PALLET_ID` be also removed?

**Review by lemunozm — DISMISSED** [review]
Minor comments. LGTM! Thanks for this

**Review by cdamian — DISMISSED** [review]
Thanks for taking care of this!

**Review by lemunozm — APPROVED** [review]

**Review by cdamian — APPROVED** [review]

**Pull request merged**